lib/spack/spack/binary_distribution.py

```diff
@@ -389,6 +412,11 @@ def make_package_placeholder(workdir, allow_root):
         cur_path_names.append(os.path.join(workdir, filename))
     relocate.make_binary_placeholder(cur_path_names, allow_root)
 
+    cur_path_names = list()
+    for filename in buildinfo['relocate_links']:
```

Comment:
I hit a bug here with an older buildcache. You need to set an empty list if relocate_links is not available.  Something like

filenames = buildinfo.get('relocate_links',False)
if filenames:
    for filename in filenames:

Reply:
Good catch. Fixed everywhere

Reply:
But don't do it like @gartung's snippet -- do this:

```python
filenames = buildinfo.get('relocate_links', [])
for filename in filenames:
    ...
```